pallets/runtime/tests/src/confidential_asset_test.rs

```diff
@@ -201,60 +235,149 @@ fn issuers_can_create_and_rename_confidential_tokens() {
+            0,          // Transaction id. Not important in this test.
+            [10u8; 32], // seed
+            &token_names[1][..],
+            valid_asset_ids,
+        );
 
-        // These are the encryptions keys used by MERCAT and are different from the signing keys
-        // that Polymesh uses.
-        let mut rng = StdRng::from_seed([10u8; 32]);
-        let elg_secret = ElgamalSecretKey::new(Scalar::random(&mut rng));
-        let elg_pub = elg_secret.get_public_key();
-        let enc_keys = EncryptionKeys {
-            pblc: elg_pub.into(),
-            scrt: elg_secret.into(),
-        };
+        ConfidentialAsset::validate_mercat_account(
+            Origin::signed(owner),
+            mercat_account_tx.clone(),
+        )
+        .unwrap();
 
-        // Positive test: use one of the registered asset ids.
-        let asset_id = AssetId {
-            id: *Ticker::try_from(&token_names[1][..]).unwrap().as_bytes(),
-        };
-        let asset_id_witness = CommitmentWitness::from((asset_id.clone().into(), &mut rng));
-        let scrt_account = SecAccount {
-            enc_keys,
-            asset_id_witness,
+        // ------------- START: Computations that will happen in Alice's Wallet ----------
+        let amount: u32 = token.total_supply; // mercat amounts are 32 bit integers.
+        let mut rng = StdRng::from_seed([42u8; 32]);
+        let issuer_account = Account {
+            scrt: scrt_account,
+            pblc: mercat_account_tx.pub_account.clone(),
         };
 
-        let valid_asset_ids = convert_asset_ids(valid_asset_ids);
-        let account_id = 0;
-        let mercat_tx_id = 0;
-        let mercat_account_tx = AccountCreator {}
-            .create(
-                mercat_tx_id,
-                &scrt_account,
-                &valid_asset_ids,
-                account_id,
+        let initialized_asset_tx = AssetIssuer {}
+            .initialize_asset_transaction(
+                1, // Transaction id. Not important in this test.
+                &issuer_account,
+                &[],
+                amount,
                 &mut rng,
             )
             .unwrap();
 
         // ------------- END: Computations that will happen in the Wallet ----------
 
+        // Wallet submits the transaction to the chain for verification.
+        ConfidentialAsset::mint_confidential_asset(
+            Origin::signed(owner),
+            ticker,
+            amount.into(), // convert to u128
+            initialized_asset_tx,
+        )
+        .unwrap();
+
+        // ------------------------- Ensuring that the asset details are set correctly
+        // Check the update investor count for the newly created asset.
+        //assert_eq!(Statistics::investor_count_per_asset(ticker), 1); // TODO not sure why this fails!
+
+        // A correct entry is added.
+        // assert_eq!(Asset::token_details(ticker), token); //  TODO: this one causes a compile error!
```

Comment:
Uncommenting it throws:
```
error[E0308]: mismatched types
   --> pallets/runtime/tests/src/confidential_asset_test.rs:333:9
    |
333 |         assert_eq!(Asset::token_details(ticker), token); //  TODO: this one causes a compile error!
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u128`, found `u32`
    |
    = note: expected struct `polymesh_primitives::SecurityToken<u128>`
               found struct `polymesh_primitives::SecurityToken<u32>`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
The total supply field in `Asset::token_details` is u128 but in your `token` struct, it's u32. Changing your total supply to u128 will fix this issue but I'm not sure if that messes up your other book keeping. 

I guess it should be fine to keep the total supply in u128 since it's not encrypted.